sdk/core/azure-core/src/main/java/com/azure/core/models/ResponseInnerError.java

```diff
@@ -56,4 +60,95 @@ public ResponseInnerError setInnerError(ResponseInnerError innerError) {
         this.innerError = innerError;
         return this;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) {
+        jsonWriter.writeStartObject();
+
+        jsonWriter.writeStringField("code", code)
+            .writeFieldName("innererror");
+
+        if (innerError != null) {
+            innerError.toJson(jsonWriter);
+        } else {
+            jsonWriter.writeNull();
+        }
+
+        // Always flush at the end of writing an object.
+        return jsonWriter.writeEndObject()
+            .flush();
+    }
+
+    /**
+     * Creates an instance of {@link ResponseInnerError} by reading the {@link JsonReader}.
+     *
+     * @param jsonReader The {@link JsonReader} that will be read.
+     * @return An instance of {@link ResponseInnerError} if the {@link JsonReader} is pointing to
+     * {@link ResponseInnerError} JSON content, or null if it is pointing to {@link JsonToken#NULL}.
+     * @throws IllegalStateException If the {@link JsonReader} wasn't pointing to the correct {@link JsonToken} when
+     * passed.
+     */
+    public static ResponseInnerError fromJson(JsonReader jsonReader) {
```

Comment:
for the future: I guess we'll want checkstyle/APIView check that `JsonCapable` implementations must also have  `public static T fromJson(JsonReader jsonReader)`

Reply:
Yeah, I want to mull over ways to enforce this and Checkstyle is a good beginning point